src/twisted/conch/scripts/ckeygen.py

```diff
@@ -44,7 +44,7 @@ class GeneralOptions(usage.Options):
                 ['showpub', 'y', 'Read private key file and print public key.']]
 
     compData = usage.Completions(
-        optActions={"type": usage.CompleteList(["rsa", "dsa"])})
+        optActions={"type": usage.CompleteList(["rsa", "dsa", "ecdsa"])})
```

Comment:
There's three different places the list "rsa", "dsa", "ecdsa" appears. This seems to be RY 😄 

Would it be better to have one place that does it?

Somewhere at the top of the file:

```
_supported_types = dict() 
def _support(tpname):
       def decorate(func):
              _supported_types[tpname] = func
              return func
    return tpname
```

and then:

```
@_support('ecdsa')
def generateECDSAkey(options):
      pass # code here
```

Here you would have

```
optActions = {'type': usage.CompleteList(list(_supported_types.keys())) }
```

below you would have

```
if options['type'] in _supported_types:
    _supported_types[option['type']](options)
else:
    sys.exit('Key type was %s, must be one of %s',
                  options['type'], ', '.join(_supported_list.keys()))
```

Especially in the wake of people starting to worry about post-quantum cryptography, I think it's important to make it more straight forward to add more types (all three listed here, as they are based on abelian groups, are vulnerable to Shor's algorithm, so we *know* that nothing here is quantum-resistant).